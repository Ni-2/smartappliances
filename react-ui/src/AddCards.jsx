import React from 'react';
import Card from './Card';
import Nav from './Nav';

export default function AddCards(props) {
  const { cards, toggleToAppliance, addNewAppliance } = props;
  return (
    <>
      <Nav tabName="My Smart Appliances" />
      <main className="cards">
        {cards &&
          Object.entries(cards).map(([id, value]) => (
            <Card key={id} id={id} value={value} toggleToAppliance={toggleToAppliance} />
          ))}
        <div className="card">
          <div className="card-body flex">
            <button className="btn btn-plus" onClick={() => addNewAppliance(true)}>
              <p className="card-text plus">+</p>
              <p className="card-text plus-add">Add new appliance</p>
            </button>
          </div>
        </div>
      </main>
    </>
  );
}
