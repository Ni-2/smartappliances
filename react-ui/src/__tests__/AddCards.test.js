import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { promises as fsp } from 'fs';
import path from 'path';
import AddCards from '../AddCards';

const getFixturePath = (filename) => path.resolve(__dirname, '__fixtures__', filename);
const readFixtureFile = (filename) => fsp.readFile(getFixturePath(filename), 'utf-8');

let cards;

beforeAll(async () => {
  const cardsStringified = await readFixtureFile('cards-101-201-301.json');
  cards = JSON.parse(cardsStringified);
});

it('renders Cards without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<AddCards />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders page with 3 cards and [+]', () => {
  const component = renderer.create(
    <AddCards
      cards={cards}
      toggleToAppliance={() => 'function'}
      addNewAppliance={() => 'function'}
    />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
