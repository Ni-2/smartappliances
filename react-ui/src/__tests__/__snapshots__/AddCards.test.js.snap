// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders page with 3 cards and [+] 1`] = `
Array [
  <nav
    aria-label="breadcrumb"
    className="breadcrumb-main"
  >
    <ol
      className="breadcrumb"
    >
      <li
        aria-current="page"
        className="breadcrumb-item active"
      >
        My Smart Appliances
      </li>
    </ol>
  </nav>,
  <main
    className="cards"
  >
    <div
      className="card"
    >
      <div
        className="card-img-container"
      >
        <img
          alt="Washing Machine"
          className="card-img-top"
          src="static/images/washing_machine_PerfectCare_800_EW8F1R48B_1.jpg"
        />
      </div>
      <div
        className="card-body"
      >
        <button
          className="btn btn-link my-btn-card"
          onClick={[Function]}
        >
          <h4
            className="card-title"
          >
            Washing Machine
          </h4>
        </button>
        <p
          className="text"
        >
          <b>
            description: 
          </b>
          in my bathroom
        </p>
        <p
          className="card-text"
        >
          <b>
            state: 
          </b>
          <span
            className="green"
          >
            available
          </span>
        </p>
      </div>
    </div>
    <div
      className="card"
    >
      <div
        className="card-img-container"
      >
        <img
          alt="Oven"
          className="card-img-top"
          src="static/images/oven_Intuit_900_PRO_KOAAS31WT_1.png"
        />
      </div>
      <div
        className="card-body"
      >
        <button
          className="btn btn-link my-btn-card"
          onClick={[Function]}
        >
          <h4
            className="card-title"
          >
            Oven
          </h4>
        </button>
        <p
          className="card-text"
        >
          <b>
            state: 
          </b>
          <span
            className="green"
          >
            available
          </span>
        </p>
      </div>
    </div>
    <div
      className="card"
    >
      <div
        className="card-img-container"
      >
        <img
          alt="Coffee Maker"
          className="card-img-top"
          src="static/images/coffee_machine_KBC65T_1.png"
        />
      </div>
      <div
        className="card-body"
      >
        <button
          className="btn btn-link my-btn-card"
          onClick={[Function]}
        >
          <h4
            className="card-title"
          >
            Coffee Maker
          </h4>
        </button>
        <p
          className="card-text"
        >
          <b>
            state: 
          </b>
          <span
            className="green"
          >
            available
          </span>
        </p>
      </div>
    </div>
    <div
      className="card"
    >
      <div
        className="card-body flex"
      >
        <button
          className="btn btn-plus"
          onClick={[Function]}
        >
          <p
            className="card-text plus"
          >
            +
          </p>
          <p
            className="card-text plus-add"
          >
            Add new appliance
          </p>
        </button>
      </div>
    </div>
  </main>,
]
`;
